include/exadg/convection_diffusion/driver.cpp

```diff
@@ -252,16 +254,28 @@ Driver<dim, Number>::solve()
              application->get_parameters().amr_data.trigger_every_n_time_steps,
              time_integrator->get_number_of_time_steps()))
         {
-          // AMR is only implemented for implicit timestepping.
-          std::shared_ptr<TimeIntBDF<dim, Number>> bdf_time_integrator =
-            std::dynamic_pointer_cast<TimeIntBDF<dim, Number>>(time_integrator);
-
-          AssertThrow(bdf_time_integrator.get(),
-                      dealii::ExcMessage("Adaptive mesh refinement only implemented"
-                                         " for implicit time integration."));
-
-          mark_cells_coarsening_and_refinement(*grid->triangulation,
-                                               bdf_time_integrator->get_solution_np());
+          // Mark cells for coarsening and refinement based on the most recent solution.
```

Comment:
introducing `TimeIntBase::get_solution_np()` would lead to:
`
mark_cells_for_coarsening_and_refinement(*grid->triangulation, time_integrator->get_solution_np());
`